src/CloudFoundry.WinDEA/StagingInstance.cs

```diff
@@ -265,34 +267,226 @@ public void GetBuildpack(StagingStartMessageRequest message, string gitPath, str
+
+                        }
                     }
-                    else
+                }
+
+                if (this.Buildpack == null)
+                {
+                    foreach (string dir in systemBuildpacks)
                     {
-                        Directory.Delete(Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        DEAUtilities.DirectoryCopy(dir, Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        Buildpack bp = new Buildpack(Path.Combine(this.Workspace.TempDir, "buildpack"), Path.Combine(this.Workspace.StagedDir, "app"), this.Workspace.Cache, this.Workspace.StagingLogPath);
+                        bool success = bp.Detect(this.Container);
+                        if (success)
+                        {
+                            this.Buildpack = bp;
+                            break;
+                        }
+                        else
+                        {
+                            Directory.Delete(Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        }
                     }
                 }
 
+
                 if (this.Buildpack == null)
                 {
                     throw new Exception("Unable to detect a supported application type");
                 }
                 Logger.Info("Staging task {0}: Detected buildpack {1}", this.Properties.TaskId, this.Buildpack.Name);
             }
+
             this.Properties.DetectedBuildpack = this.Buildpack.Name;
         }
 
+        public static bool IsWindowsCompatibleBuildpack(string buildpackDir)
+        {
+            var bins = new string[] { "detect", "compile", "release" };
+            string[] pathExt = Environment.GetEnvironmentVariable("PATHEXT").Split(';');
+
+            foreach (var app in bins)
+            {
+                bool found = false;
+                foreach (string ext in pathExt)
+                {
+                    if (File.Exists(Path.Combine(buildpackDir, "bin", app + ext)))
+                    {
+                        found = true;
+                    }
+                }
+                if (!found)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public void DownloadAdminBuildpack(StagingStartRequestAdminBuildpack adminBuildpack, string adminBuildpackDir)
+        {
+            string archiveFile = Path.Combine(adminBuildpackDir, adminBuildpack.Key + ".zip");
+            string destDir = Path.Combine(adminBuildpackDir, adminBuildpack.Key);
+
+            if (Directory.Exists(destDir))
+            {
+                Logger.Info("Skipping download for admin buildpack {0} because it was previously downloaded in {1} ", adminBuildpack.Key, destDir);
+                return;
+            }
+
+            Directory.CreateDirectory(destDir);
+
+            WebClient client = new WebClient();
```

Comment:
WebClient is IDisposable.
CodeAnalysis/StyleCop should be complaining about this.